enterprise/api/v1/views.py

```diff
@@ -213,6 +213,110 @@ def course_enrollments(self, request, pk):
 
         return Response(serializer.errors, status=HTTP_400_BAD_REQUEST)
 
+    @action(detail=True, methods=['post'], permission_classes=[permissions.IsAuthenticated])
+    @permission_required('enterprise.can_enroll_learners', fn=lambda request, pk: pk)
+    # pylint: disable=invalid-name,unused-argument
+    def enroll_learners_in_courses(self, request, pk):
+        """
+        Creates a set of licensed enterprise_learners by bulk enrolling them in the specified course.
+
+        Note: This endpoint isn't transactional in that if an enrollment fails, there can potentially be a scenario
+        with a subset of users enrolled. Requested learners' eligibility to enroll are expected to be pre-vetted.
+
+        Expected params:
+            - license_info (Dict): user emails as keys and the license UUIDs associated with each of the courses the
+            user's being enrolled in as values. Note, each user is expected to have exactly one license per course
+            listed in the 'courses' parameter.
+            Example:
+                license_info: {
+                    'newuser@test.com': {
+                        'course-v1:edX+DemoX+Demo_Course': '5b77bdbade7b4fcb838f8111b68e18ae'
+                    },
+                    ...
+                }
+
+            - courses (Dict): course run keys as keys and the course mode as the value.
+            Example:
+                'courses': {
+                    'course-v1:edX+DemoX+Demo_Course': 'verified',
+                    ...
+                }
+        """
+        enterprise_customer = self.get_object()
+        serializer = serializers.EnterpriseCustomerBulkSubscriptionEnrollmentsSerializer(
+            data=request.data,
+            context={
+                'enterprise_customer': enterprise_customer,
+                'request_user': request.user,
+            }
+        )
+        if serializer.is_valid(raise_exception=True):
```

Comment:
Change needed: There is a pattern here I would like you to change to and start using in general (called the Bouncer Pattern https://softwareengineering.stackexchange.com/questions/18454/should-i-return-from-a-function-early-or-use-an-if-statement)  which is to invert your logic and _exit_ right here if this fails instead of nesting the next 80 lines of code. It makes it clearer what's going on.

 When you make this change, also remember to capture the serializer errors and send them back, which is missing in the call on line 318